### PR TITLE
plugins: Add SpammingAlert plugin

### PR DESCRIPTION
**Pull request by wyj1026**
Alert the user that they are spamming.
See: https://github.com/coala/corobo/issues/482 and https://gitlab.com/coala/cobot/issues/16

# Reviewers Checklist

- [ ] Appropriate logging is done.
- [ ] Appropriate error responses.
- [ ] Handle every possible exception.
- [ ] Make sure there is a docstring in the command functions. Hint: Lookout for
  `botcmd` and `re_botcmd` decorators.
- [ ] See that 100% coverage is there.
- [ ] See to it that mocking is not done where it is not necessary.

### CODE REVIEW

File: plugins/spam.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    def callback_message(self, msg):
+        """
+        Alert the user that his/her message is too long.
+        """
+
+        if len(msg.body) > self.config['MAX_MSG_LEN'] or \
+           msg.body.count('\n') >= self.config['MAX_LINES']:
+            self.send(msg.frm,
+                      "@{} it seems you're spamming. Pay attention to your "
+                      "wrods, please.".format(msg.frm.nick)
```
[the remark this answers is not in the document]
Comment (wyj1026) [reply plugins/spam.py]: fixed :)

**Comment by jayvdb** [comment]
semaphore CI is failing because of missing coverage.  You need to add one more test method, to reach one more line.  See https://semaphoreci.com/coala/corobo/branches/pull-request-489/builds/4

**Comment by wyj1026** [comment]
Thanks for the feedback, tests updated and CI passed. : )

File: plugins/spam.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+from plugins import constants
+
+
+class SpammingAlert(BotPlugin):
+    """
+    A callback to alert the user that they might be spamming.
+    """
+
+    CONFIG_TEMPLATE = {'MAX_MSG_LEN': constants.MAX_MSG_LEN,
```
Comment (jayvdb) [inline plugins/spam.py]: put the dict items on the next line, and end each line with a `,`

File: plugins/spam.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+from itertools import chain
+
+from errbot import BotPlugin
+
+from plugins import constants
+
+
+class SpammingAlert(BotPlugin):
+    """
+    A callback to alert the user that they might be spamming.
```
Comment (jayvdb) [inline plugins/spam.py]: s/callback/plugin/

The callback is the `callback_message` method below.

You could say

> A plugin which uses a callback to alert...

But line length might be a problem with that.

File: plugins/spam.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    def callback_message(self, msg):
+        """
+        Alert the user that his/her message is too long.
+        """
+
+        if (len(msg.body) > self.config['MAX_MSG_LEN'] or
+                msg.body.count('\n') >= self.config['MAX_LINES']):
+            self.send(msg.frm,
+                      "@{} it seems you're spamming. Pay attention to your "
```
Comment (jayvdb) [inline plugins/spam.py]: please put this message in a template.  See 11ce167b47c96c35

**Comment by wyj1026** [comment]
All fixed, anything more I need to do?

File: plugins/spam.py (hunk 3 above)
Comment (wyj1026) [reply plugins/spam.py]: I've made changes, but it still says changes are requested, did I miss something?

**Comment by wyj1026** [comment]
@jayvdb Could you please review it? Thanks.

File: plugins/spam.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    def get_configuration_template(self):
+        return self.CONFIG_TEMPLATE
+
+    def configure(self, configuration):
+        """
+        Enable partial configuration.
+        """
+
+        if configuration is not None and configuration != {}:
```
Comment (nemaniarjun) [inline plugins/spam.py]: This could be replaced by "if configuration:"

**Comment by nemaniarjun** [comment]
Just that one comment that needs to be addressed. Also a rebase is needed :)

**Comment by nvzard** [comment]
@wyj1026 can you please make the requested changes?

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by meetmangukiya** [comment]
@gitmate-bot rebase

**Comment by meetmangukiya** [comment]
ack https://github.com/coala/corobo/pull/489/commits/fc9ba1e9e938938305b90ec6fb30361071482e0a

**Comment by meetmangukiya** [comment]
@gitmate-bot ff

**Pull request merged**